mypy/newsemanal/semanal.py

```diff
@@ -1882,7 +1853,12 @@ def visit_assignment_stmt(self, s: AssignmentStmt) -> None:
         self.check_and_set_up_type_alias(s)
         self.newtype_analyzer.process_newtype_declaration(s)
         self.process_typevar_declaration(s)
-        self.named_tuple_analyzer.process_namedtuple_definition(s, self.is_func_scope())
+        if isinstance(s.rvalue, CallExpr) and isinstance(s.rvalue.analyzed, NamedTupleExpr):
+            existing_info = s.rvalue.analyzed.info  # type: Optional[TypeInfo]
```

Comment:
I think that it would be better to do nothing on subsequent iterations if we've already successfully analyzed a named tuple definition. Analysis should only be successful if everything can be bound, and in that case the second iteration should produce the same result. However, in case there is a bug that causes the first iteration to produce an incomplete or invalid type, the second iteration could hide this bug in many cases, making the bug harder to find. Additionally, the logic would be a bit simpler.

We can't use this with normal class definitions though, since base types can be incomplete after the first iteration.